src/Stratis.Bitcoin.Features.Wallet/WalletManager.cs

```diff
@@ -1003,6 +1010,42 @@ public bool ProcessTransaction(Transaction transaction, int? blockHeight = null,
 
             lock (this.lockObject)
             {
+                // Do a pre-scan of the transaction inputs to see if they're used in other transactions already
+                foreach (TxIn input in transaction.Inputs)
+                {
+                    // See if this input is being used by another transaction.
+                    if (this.inputLookup.TryGetValue(input.PrevOut, out WalletIndexData indexData))
+                    {
+                        if (indexData.TransactionData.Id == hash)
+                            continue; // It's the same transaction, ignore.
+
+                        // Incoming transaction confirmed, existing transaction unconfirmed, delete existing
+                        if ((block != null) && (indexData.TransactionData.BlockHash == null))
```

Comment:
Perhaps rename `block` to `confirmedBlock`? So that we know that we are working with a transaction that is in a confirmed block?

Reply:
It sort of implies it already - this method gets called both for mempool transactions and transactions found in a block (and you don't necessarily know which order these will occur or if it will even be called under both scenarios).

Therefore if `block` is non-`null` the transaction must be confirmed (unless there is a very inopportune reorg), and if it is `null` it probably isn't (there are potential race conditions with the mempool). So renaming the variable doesn't really add clarity I think, since the contents of the variable tells you everything you need to know.